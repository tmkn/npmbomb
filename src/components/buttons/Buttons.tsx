/** @jsx jsx */
import { jsx, css } from "@emotion/core";

import { mq } from "../../css";

export const PrimaryButton: React.FC = ({ children }) => {
    return <button>{children}</button>;
};
