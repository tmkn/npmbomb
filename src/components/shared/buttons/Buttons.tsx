/** @jsx jsx */
import { jsx, css } from "@emotion/core";

import { mq, sansSerifFont, primaryColor, primaryColorLight } from "../../../css";

const buttonStyle = css({
    [mq[0]]: {
        display: "flex",
        flex: 1,
        backgroundColor: `${primaryColorLight}`,
        color: "white",
        fontSize: "1rem",
        fontFamily: `"${sansSerifFont}"`,
        justifyContent: "center",
        padding: "0.5rem 0",
        cursor: "pointer",
        border: `none`,
        ":hover": {
            backgroundColor: `${primaryColor}`
        }
    },
    [mq[1]]: {
        width: "10vw",
        flex: "unset"
    },
    [mq[2]]: {}
});

interface IPrimaryButtonProps {
    onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
    disabled?: boolean;
}

export const PrimaryButton: React.FC<IPrimaryButtonProps> = ({ children, onClick, disabled }) => {
    const isDisabled = disabled ?? false;

    const disabledStyle = css({
        [mq[0]]: {
            backgroundColor: "#e8eaf6",
            ":hover": {
                backgroundColor: "#e8eaf6"
            }
        }
    });

    const styles = [buttonStyle];
    if (isDisabled) styles.push(disabledStyle);

    function onClickImpl(e: React.MouseEvent<HTMLButtonElement>) {
        if (!isDisabled && onClick) onClick(e);
    }

    return (
        <button css={styles} onClick={onClickImpl}>
            {children}
        </button>
    );
};
