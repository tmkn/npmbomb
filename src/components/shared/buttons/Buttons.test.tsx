import React from "react";
import { render, fireEvent, waitForElement, wait } from "@testing-library/react";

import { PrimaryButton } from "./Buttons";

describe("<PrimaryButton />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<PrimaryButton>Label</PrimaryButton>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("matches disabled snapshot", () => {
        const { asFragment } = render(<PrimaryButton disabled>Label</PrimaryButton>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("executes callback", async () => {
        const callback = jest.fn();
        const { getByText } = render(<PrimaryButton onClick={callback}>Label</PrimaryButton>);

        fireEvent(
            getByText("Label"),
            new MouseEvent("click", {
                bubbles: true,
                cancelable: true
            })
        );

        expect(callback).toHaveBeenCalled();
    });

    test("doesn't executes callback", async () => {
        const callback = jest.fn();
        const { getByText } = render(
            <PrimaryButton disabled onClick={callback}>
                Label
            </PrimaryButton>
        );

        fireEvent(
            getByText("Label"),
            new MouseEvent("click", {
                bubbles: true,
                cancelable: true
            })
        );

        expect(callback).not.toHaveBeenCalled();
    });
});
