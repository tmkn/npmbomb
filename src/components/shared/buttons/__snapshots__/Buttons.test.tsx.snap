// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> matches disabled snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-x9wjhs"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`<PrimaryButton /> matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-exwq4w"
  >
    Label
  </button>
</DocumentFragment>
`;
